lib/api/v1/health.rb

```diff
@@ -14,14 +14,16 @@ class Health < Grape::API
 
       resource :health do
         desc "Returns hash of metrics",
-             tags:   ["health"],
-             detail: "Returns general metrics on the health of the system"
+             entity:  API::Entities::Health,
+             tags:    ["health"],
+             detail:  "Returns general metrics on the health of the system",
+             failure: [[503, "Some of the required services are unhealthy"]]
 
         get do
           response, success = ::Portus::Health.check
           code = success ? 200 : 503
           status code
-          response
+          present response
```

Comment:
Should we use `with: API::Entities::Health` here? I guess the object structure is the same but if it change it in the future we wouldn't be inconsistent with the documentation since it forces to use the entity.

Grape could be smarter and if we omit the `with`, then use whatever is declared in the `entity` attribute.

Reply:
> Should we use with: API::Entities::Health here? I guess the object structure is the same but if it change it in the future we wouldn't be inconsistent with the documentation since it forces to use the entity.

I tried that, but apparently Grape entity is not capable of matching hash keys with attributes...

> Grape could be smarter and if we omit the with, then use whatever is declared in the entity attribute.

Well, the fact that it doesn't try to be too smart is what allows us to work-around the flaw that I mentioned 😉

Reply:
> I tried that, but apparently Grape entity is not capable of matching hash keys with attributes...

That's sad.

> Well, the fact that it doesn't try to be too smart is what allows us to work-around the flaw that I mentioned 😉

I know it gives more flexibility but I guess it's more common to match the entity than to not use it. I'm sure people already suggested that to them. I was just thinking out loud again.